pkg/ingester/ingester_ring.go

```diff
@@ -75,15 +76,22 @@ func (cfg *RingConfig) Validate() error {
 		if cfg.TokensFilePath != "" {
 			return fmt.Errorf("%q token generation strategy requires %q to be empty", spreadMinimizingTokenGeneration, tokensFilePathFlag)
 		}
-		_, err := ring.NewSpreadMinimizingTokenGenerator(cfg.InstanceID, cfg.InstanceZone, cfg.SpreadMinimizingZones, cfg.SpreadMinimizingJoinRingInOrder)
-		return err
+
+		if _, err := ring.NewSpreadMinimizingTokenGenerator(cfg.InstanceID, cfg.InstanceZone, cfg.SpreadMinimizingZones, cfg.SpreadMinimizingJoinRingInOrder); err != nil {
+			return err
+		}
 	}
 
-	// at this point cfg.TokenGenerationStrategy is not spreadMinimizingTokenGeneration
-	if cfg.SpreadMinimizingJoinRingInOrder {
+	if cfg.TokenGenerationStrategy != spreadMinimizingTokenGeneration && cfg.SpreadMinimizingJoinRingInOrder {
 		return fmt.Errorf("%q must be false when using %q token generation strategy", spreadMinimizingJoinRingInOrderFlag, cfg.TokenGenerationStrategy)
 	}
 
+	// Ingest storage requires the zone to be configured for ingesters, because on the read path we always use
+	// zone-aware replication tracker to compute the quorum.
+	if ingestStorageEnabled && cfg.InstanceZone == "" {
+		return errors.New("the ingester instance zone must be configured when running Mimir with the ingest storage")
```

Comment:
do we have a more exact cli config name we can give in this error, so it's easier to fix? maybe we can mention both the ingest storage and the ring zone config

Reply:
Sure, done in https://github.com/grafana/mimir/pull/7432/commits/5dad7ba55af250b7fbc54c98d8ac6b3bc646a0f9